summarize_test_results.py

```diff
@@ -531,6 +584,68 @@ def format_alerts(summary, embed=True, file_out=None):
         print("EOF", file=file_out)
 
 
+def compute_semaphore(success_percent, embed=True):
+    """create a semaphore light summarizing the success percent.
+    If set to `embed`, an emoji will be used. Else, a textual representation
+    of a Slack emoji is used.
+    """
+    if embed:
+        if success_percent >= 95:
+            return "🟢"
+        elif success_percent >= 60:
+            return "🟡"
+        else:
+            return "🔴"
+    else:
+        if success_percent >= 95:
+            return ":large_green_circle:"
+        elif success_percent >= 60:
+            return ":large_yellow_circle:"
+        else:
+            return ":red_circle:"
+
+
+def compute_thermometer_on_metric(summary, metric, embed=True):
+    """computes a summary per item in the metric, with the success percentage
+    and a color coding based on said percentage
+    """
+
+    output = f"{metric_name(metric)} thermometer:\n\n"
+    for bucket_hits in summary[metric]["failed"].items():
```

Comment:
We just found a bug here, when @NiccoloFei ran a CI/CD workflow with this branch (latest SHA)
https://github.com/EnterpriseDB/cloudnative-pg/actions/runs/9369542163
Since we iterate over the `summary[metric]["failed"]`, we will skip over items that had no failures.
We should still get a printout even if there were zero failures.
Proboably better iterate over `summary[metric]["total"]`
@jlong49